js/stock-exchange.js

```diff
@@ -0,0 +1,153 @@
+const STOCKS_BASE_URL =
+    "https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com";
+const STOCKS_SEARCH_LIMIT = 10;
+const SYMBOL_PREFIX = "$";
+const API_PROFILE_COMPANY_LIMIT = 3;
+
+let search = document.getElementById("btnStockSearch");
+let inputStockSearch = document.getElementById("inputStockSearch");
+var lastSymbol = "";
+
+inputStockSearch.addEventListener("keypress", function (event) {
+    if (event.key === "Enter") {
+        event.preventDefault();
+
+        // Enter key on input works as button click, makes Fibonacci calculation
+        document.getElementById("btnStockSearch").click();
+    }
+});
+
+search.addEventListener("click", function () {
+
+    removeStockProfileInfo();
+
+    let searchText = document.getElementById("inputStockSearch").value;
+
+    if (searchText !== "") {
+        let searchURL = `${STOCKS_BASE_URL}/api/v3/search?query=${searchText}&amp;limit=${STOCKS_SEARCH_LIMIT}&amp;exchange=NASDAQ`;
+        let stockList = document.getElementById("companyProfile");
+
+        if (stockList) {
+            stockList.replaceChildren();
+            document.getElementById("stockPlot").style.display = "none";
+        }
+
+        showLoading();
+
+        fetch(searchURL)
```

Comment:
extract this code to another function instead of writing such a long code inside an anonymous function of event listener. The functionality will be the same but the design of the code will be much better.